### ISSUE TITLE
Setting Priority to filter SalesPrices in Codeunit7000 in Function FindSalesPrice

### ISSUE THREAD

**Issue by krishnaGottuparthi** [microsoft/ALAppExtensions#2706]
Is it possible to filter SalesPrices based  on priority levels or hierarchy defined for "Sales Type".
This is like an enhancement in which function **FindSalesPrice** should be modified.
For example:
We have SalesPrices mapped for "Sales Type"::"All Customers" , "Sales Type"::Customer and "Sales Type"::"Customer Price Group" as follows
![image](https://user-images.githubusercontent.com/35251581/58892815-945d2380-870c-11e9-8edd-5cfad3afa030.png)

Standard NAV calculates the bestprice as **5** without considering "Sales Type" since its the least price.

Here,we need to set some priority to first check for **"Sales Type"::Customer** and restrict further check for other Sales Types if Prices are found.
![image](https://user-images.githubusercontent.com/35251581/58893730-519c4b00-870e-11e9-849d-6c72beb7ef3e.png)

If there aren't any Prices mapped for "Sales Type"::Customer then system should check for other "Sales Type"::"Customer Price Group"
If still any Prices are not found for "Sales Type"::"Customer Price Group" then further filtering should be done for "Sales Type"::"All Customers"

This is more like setting priority of filtering Sales Prices in Table **FromSalesPrice**  based on the priority levels defined in a new parameter for Function **FindSalesPrice**

Also there should be possibility to make this work with combination also.Like the way standard works now.
There should be something like a boolean kind of parameter for this function which indicates whether this function should filter all sales types or individually for specific sales types

In other words,this parameter differentiates existing standard functionality now and new enhancement we requested

Note:If this can be included then similar kind of functionality needs to be incorporated Discounts side also.
i.e. Sales Line Discounts
![image](https://user-images.githubusercontent.com/35251581/59025927-e9b34500-8872-11e9-9c8e-146780b4d167.png)

Please check and let us know

**Comment by AlexanderYakunin** [comment]
We have implemented new pricing framework in 17.1, so I'm closing this old request for now.
